drivers/SmartThings/jbl/src/jbl/api.lua

```diff
@@ -0,0 +1,138 @@
+local log = require "log"
+local json = require "st.json"
+local RestClient = require "lunchbox.rest"
+local utils = require "utils"
+local cosock = require "cosock"
+
+local jbl_api = {}
+jbl_api.__index = jbl_api
+
+local CREDENTIAL_TIME_OUT_SECONDS = 30
+
+local SSL_CONFIG = {
+    mode = "client",
+    protocol = "any",
+    verify = "peer",
+    options = "all",
+    cafile="./selfSignedRoot.crt"
+}
+
+local ADDITIONAL_HEADERS = {
+    ["Accept"] = "application/json",
+    ["Content-Type"] = "application/json",
+}
+
+function jbl_api.labeled_socket_builder(label)
+    local socket_builder = utils.labeled_socket_builder(label, SSL_CONFIG)
+    return socket_builder
+end
+
+local function get_base_url(device_ip)
+    return "https://" .. device_ip .. ":4443"
+end
+
+local function process_rest_response(response, err, partial)
+    if err ~= nil then
+        return response, err, nil
+    elseif response ~= nil then
+        local _, decoded_json = pcall(json.decode, response:get_body())
```

Comment:
The `pcall` result should be checked before using the returned value.

Reply:
done